edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.IoTHub/EdgeAgentConnection.cs

```diff
@@ -184,14 +184,14 @@ async Task RefreshTwinAsync()
                 });
         }
 
-        async Task<Option<Twin>> GetTwinFromIoTHub()
+        async Task<Option<Twin>> GetTwinFromIoTHub(bool forceNewModuleClient = false)
         {
             try
             {
                 async Task<Twin> GetTwinFunc()
                 {
-                    Events.GettingModuleClient();
-                    IModuleClient moduleClient = await this.moduleConnection.GetOrCreateModuleClient();
+                    Events.GettingModuleClient(forceNewModuleClient);
+                    IModuleClient moduleClient = await this.moduleConnection.GetOrCreateModuleClient(forceNewModuleClient);
```

Comment:
This will cause the module client to be created every time in the retry operation (up to 3 times). 
We should do it only once. Maybe it may be easier to close the moduleClient in the code here and let GetOrCreateModuleClient create a new one.

Reply:
If we come to this point, it means something happens to the module client other than connection timeout.  If that is the case, I expect it should be succeeded in 1st call and no retry should be needed.  If retry really happens, it is no harmful to re-create a module client.  I think this is more simple, isn't it?